UGENE-7325. Make the "PCR primer design" tab visible for DNA sequence only

evelinabis opened the pull request:
Tab is invisible to RNA, Amino, raw alphabets.

Commit:
UGENE-7325. Make the "PCR primer design" tab visible for DNA sequence
only.
The widget is dynamically enabled/disabled when selecting a
nucleotide/non-nucleotide sequence.
Reconstructed UI file: added a widget to the root containing other
widgets, added a warning label.

Commit:
UGENE-7325. Make the "PCR primer design" tab visible for DNA sequence
only.
Registered PCR Primer Design tab in GUITest enum.
Added GUI test.

Commit:
UGENE-7325. Make the "PCR primer design" tab visible for DNA sequence
only.

Added the ability to filter option panel tabs by RNA.
PCR Primer Design tab is available only for sequence files containing
DNA sequence.
Added GUI test for new RNA behavior.
Simplified test_7325.
Minor changes: added newline at end of file.

Commit:
Merge branch 'pcr_primer_design_for_dna_assembly' of https://github.com/ugeneunipro/ugene into UGENE-7325

DmitriiSukhomlinov commented on src/corelibs/U2Gui/src/options_panel/OPWidgetFactory.h:
```diff
@@ -69,6 +71,9 @@ class U2GUI_EXPORT OPFactoryFilterVisitorInterface {
     virtual bool typePass(ObjectViewType factoryViewType) = 0;
     virtual bool alphabetPass(DNAAlphabetType factoryAlphabetType) = 0;
     virtual bool atLeastOneAlphabetPass(DNAAlphabetType factoryAlphabetType) = 0;
+
```
Remove the empty line

DmitriiSukhomlinov commented on src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp:
```diff
@@ -434,6 +436,82 @@ GUI_TEST_CLASS_DEFINITION(test_7212) {
     GTUtilsTaskTreeView::waitTaskFinished(os);
 }
 
+GUI_TEST_CLASS_DEFINITION(test_7325) {
```
Don't remove this test, but, take an account, that this feature doesn't need cuz it's not finished yet

DmitriiSukhomlinov commented on src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp:
```diff
@@ -42,6 +43,15 @@
 
+// When a non-nucleotide sequence is selected, the widget should be disabled. But the groupbox titles continue to be
+// black as if they enabled. This code makes them gray.
+static void makeGroupboxTittleGrayIfDisable(QGroupBox *const gb) {
```
This function is cut off from everything, It's better to move it to either the `PCRPrimerDesignForDNAAssemblyOPWidget` class or the anonimus namespace. But, as far is this function is used in the only one function, you can just simply define it as lambda.

DmitriiSukhomlinov reviewed (changes requested):
**Scenario:**

1. Open the "ex.fa"file, which is attached to the issue ([link](https://ugene.dev/tracker/secure/attachment/19951/ex.fa)).
2. Open the "PCR primer design"tab.
Expected: the tab is disabled.
3. Change focus to the first sequence
Expected: the tab has become enabled.
4. Click any "Select manually" button and start selecting on the Zoom view of the first sequence.
Expected: the corresponding fields are being changed.
Current: the corresponding fields aren't changed

Also see minor comments

Commit:
UGENE-7325. Make the "PCR primer design" tab visible for DNA sequence
only.

Changes by review.
Small code refactoring, such as moving and splitting code into separate
methods.
Fixed region manually selection.
Fixed scenario from pull request.
Added a change allowed spinboxes ranges when changing sequence length.
Added Select manually button disabling when changing active sequence.

evelinabis commented:
https://github.com/ugeneunipro/ugene/blob/9e5015a6e82696f0f62691f3eff6ba7986094d6f/src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp#L407
This is context dependent. I guess it may happen that while the task is being executed, context may change. I could dispel doubts, but I don't know how to start PCR Primer Design task.

DmitriiSukhomlinov commented:
OK, merge it now, I'll report the other issue if some bags will be found
